src/Analysis/Ast/Test/ImportTests.cs

```diff
@@ -151,5 +154,59 @@ import sys
             analysis.Should().HaveVariable("P")
                 .Which.Should().HaveMembers(@"abspath", @"dirname");
         }
+
+        [TestMethod, Priority(0)]
+        public async Task UnresolvedImport() {
+            var analysis = await GetAnalysisAsync(@"import nonexistent");
+            analysis.Should().HaveVariable("nonexistent")
+                .Which.Value.GetPythonType<IPythonModule>().ModuleType.Should().Be(ModuleType.Unresolved);
+            analysis.Diagnostics.Should().HaveCount(1);
+            var d = analysis.Diagnostics.First();
+            d.ErrorCode.Should().Be(ErrorCodes.UnresolvedImport);
+            d.SourceSpan.Should().Be(1, 8, 1, 19);
```

Comment:
Can we use `Range` instead of `SourceSpan`?

Reply:
Range is not in analysis, it is in LSP. Can theoretically move it down.

Reply:
I thought `Diagnostics` is part of LSP as well. We also have `IndexSpan`.

Reply:
`Diagnostics` yes. `DiagnosticsEntry`, no. Switching away from SourceSpan would require changing tens of tests. Analysis is using `LocationInfo` which produces `SourceSpan`.